networkx/linalg/laplacianmatrix.py

```diff
@@ -194,50 +222,112 @@ def normalized_laplacian_matrix(G, nodelist=None, weight="weight"):
     if nodelist is None:
         nodelist = list(G)
     A = nx.to_scipy_sparse_array(G, nodelist=nodelist, weight=weight, format="csr")
-    n, m = A.shape
+    n, _ = A.shape
     diags = A.sum(axis=1)
     # TODO: rm csr_array wrapper when spdiags can produce arrays
-    D = sp.sparse.csr_array(sp.sparse.spdiags(diags, 0, m, n, format="csr"))
+    D = sp.sparse.csr_array(sp.sparse.spdiags(diags, 0, n, n, format="csr"))
     L = D - A
     with np.errstate(divide="ignore"):
         diags_sqrt = 1.0 / np.sqrt(diags)
     diags_sqrt[np.isinf(diags_sqrt)] = 0
     # TODO: rm csr_array wrapper when spdiags can produce arrays
-    DH = sp.sparse.csr_array(sp.sparse.spdiags(diags_sqrt, 0, m, n, format="csr"))
+    DH = sp.sparse.csr_array(sp.sparse.spdiags(diags_sqrt, 0, n, n, format="csr"))
     return DH @ (L @ DH)
 
 
 @nx._dispatchable(edge_attrs="weight")
-def total_spanning_tree_weight(G, weight=None):
+def total_spanning_tree_weight(G, weight=None, root=None):
     """
     Returns the total weight of all spanning trees of `G`.
 
-    Kirchoff's Tree Matrix Theorem states that the determinant of any cofactor of the
-    Laplacian matrix of a graph is the number of spanning trees in the graph. For a
-    weighted Laplacian matrix, it is the sum across all spanning trees of the
-    multiplicative weight of each tree. That is, the weight of each tree is the
-    product of its edge weights.
+    Kirchoff's Tree Matrix Theorem [1]_, [2]_ states that the determinant of any
```

Comment:
I would just as soon leave the remainder of the docstring for this deprecated function as-is, but I guess since the changes are already in place it's worth just pushing on without reviewing too carefully!

Reply:
I'll leave them as they are...